src/act.core.etl/Gatherer.cs

```diff
@@ -430,20 +430,32 @@ private static string ToQueryString(IDictionary<string, string> nvc)
         {
             return "?" + string.Join("&", nvc.Select(p => $"{p.Key}={p.Value}"));
         }
-        
-        
+
+
         public async Task<int> PurgeInactiveNodes()
         {
             var date = DateTime.Today.AddDays(-7);
             var count = 0;
-            _logger.LogDebug($"Getting Nodes Deactivated prior to {date.ToShortDateString()}");
+            _logger.LogInformation($"Getting Nodes Deactivated prior to {date.ToShortDateString()}");
+
+            var nodes = await _ctx.Nodes.Inactive().Where(p => p.DeactivatedDate < date)
+                .Select(p => new { p.Fqdn, p.DeactivatedDate, p.BuildSpecification.Id })
+                .ToArrayAsync();
+
+            if (nodes.Length > 0)
+            {
+                foreach (var node in nodes)
+                {
+                    _logger.LogInformation($"Deactivated Node : fqdn {node.Fqdn}, deactivated date : {node.DeactivatedDate} , buildspecid : {node.Id}");
+                }
```

Comment:
Also do we really need a foreach to just log the information?
If this is really needed, how about doing this asynchronously!

Reply:
Yes, we need to monitor all the nodes which are been purged. This will help us troubleshoot a ACT issue. Also, this code will be reverted once we identify the issue.